docs/Contributing/Testing-and-local-development.md

```diff
@@ -81,6 +81,14 @@ REDIS_TEST=1 MYSQL_TEST=1 make test
 
 The integration tests in the `server/service` package can generate a lot of logs mixed with the test results output. To make it easier to identify a failing test in this package, you can set the `FLEET_INTEGRATION_TESTS_DISABLE_LOG=1` environment variable so that logging is disabled.
 
+The MDM integration tests are run with a random selection of software installer storage backends (local filesystem or S3/minio), and similar for the bootstrap packages storage (DB or S3/minio). You can force usage of the S3 backend by setting `FLEET_INTEGRATION_TESTS_SOFTWARE_INSTALLER_STORE=s3`. Note that `MINIO_STORAGE_TEST=1` must also be set for the S3 backend to be used.
```

Comment:
I ended up going with that approach instead of creating a whole different test suite with the S3 backend enabled, with 99% of the setup the same, and having to rewrite tests just for that or extract tests from the two suites so they're reusable... I also checked how we could change that backend dynamically in the existing MDM suite, but that's very hairy given that it's set deep down when the EE service is created and then it's hidden in a private field on that struct.

I think this makes sense given that the change of backend should not impact any behaviour at the integration level, so the same tests should have the same results and expectations. We can write a test once and have it run on both backends, the only gotcha is that if you write something that's closely related to the S3 backend, then you should explicitly run the suite with both backends at least locally, to verify it works on both (which is what I did via the env var).

I added a log when S3 backend is used so that if there's a failure and it's relevant, there's a way to see what backend was enabled.

Happy to discuss further if y'all don't like this approach!